src/pkcs1v15.rs

```diff
@@ -436,6 +439,71 @@ where
     }
 }
 
+#[cfg(feature = "algorithm-identifier")]
+impl<D> AssociatedAlgorithmIdentifier for SigningKey<D>
+where
+    D: Digest,
+{
+    type Params = AnyRef<'static>;
+
+    const ALGORITHM_IDENTIFIER: AlgorithmIdentifierRef<'static> = pkcs1::ALGORITHM_ID;
+}
+
+#[cfg(all(feature = "algorithm-identifier", feature = "sha1"))]
+impl spki::SignatureAlgorithmIdentifier for SigningKey<sha1::Sha1> {
+    type Params = AnyRef<'static>;
+
+    const SIGNATURE_ALGORITHM_IDENTIFIER: AlgorithmIdentifierRef<'static> =
+        AlgorithmIdentifierRef {
+            oid: const_oid::ObjectIdentifier::new_unwrap("1.2.840.113549.1.1.5"),
```

Comment:
@tarcieri @dignifiedquire 
Actually I'm thinking about a special `trait RsaEncryptionOid` which then can be implemented for `Digest`.
The `RsaEncryptionOid` can go into traits (Ugh), and then digest implementations can implenent them on their own.

WDYT?

Reply:
In `ecdsa` I'm mapping from the digest OIDs to the ECDSA OIDs, in a manner which could work as a `const fn` potentially. Would something like that work here?

Reply:
I thought about that, but strictly speaking I don't like the idea of handling all the mapping here. Initially I had just sha1/sha256 (which was bearable). Then it got extended to all SHA2 functions. And there exist MD4/MD5/MD2/etc. This is unlike ECDSA where there are just 4 digest functions.

Reply:
I think I'm going to split the PSS common PKCS1v15 implementation (so that it can be merged now) and leave pkcs1v15 Signature stuff as a draft.

Reply:
Alternatively you could define the carrier trait in the RSA crate and conditionally implement it when the e.g. `sha2` feature is enabled.

I wouldn't worry too much about extended digests that aren't commonly used, at least at first, particularly broken ones like MD2, MD4, and MD5. We can cross that bridge when we get there.

Reply:
Well Sha1 is broken too. It's all about the legacy and existing files. The bonus of the global `RsaSignatureOid` is that we won't have to bloat the RSA crate with tons of features.

Reply:
It already has a `sha2` feature, FWIW.

The features make thing nicer for downstream users. It's one less dependency for them to import, and it took care of all of the confusion around enabling the `oid` feature of `sha2`.

Reply:
We can reuse the "oid" ;-) I fear that selecting RSA's features might also become an issue.

Reply:
I have implemented the `SignatureAlgorithmIdentifier` for PKCS1v15 keys in a way that doesn't require `panic!`, provide compile time checks and can be migrated to generic traits/in-digest-trait implementation.